js/panel-config/json_textarea.js

```diff
@@ -32,10 +32,12 @@ export default class JSONTextArea extends Component {
   }
 
   componentWillReceiveProps({ value }) {
```

Comment:
I don't think that this is a good fix as it means that unsaved changes can still get overwritten if they are valid JSON.

The correct approach is to only change the state if the passed in value has changed from the previous value.

```js
if (value === this.props.value) return;
```

Reply:
Thanks for the advice! I have reworked this change as per your suggestion. 

However just so you are aware, I was not able to reproduce a case where valid JSON would get overwritten, even with the original implementation. Any insight as to why (purely out of curiosity)?.

Reply:
Oh, I think I know why that is the case. The moment it is valid, we call `this.props.onChange(parsed);`, which means that the parent is notified of the new value. Home Assistant will store the new value and pass it down again, making sure it persists.